### PR TITLE
Fixed PEP8 errors in doc/source/whatsnew/v0.15.*

### LINKED ISSUE #24239: DOC: Fix flake8 issues in doc/source/whatsnew/v0.15.*.rst
xref: #24177 

We didn't start validating the format of PEP8 and other code standards in the documentation examples until recently. We still have some files with errors, that we need to skip, and that we should fix, so we can start validating them.

**Two steps are required to replicate the issue**

1. Edit `setup.cfg` in the pandas home, and in the flake8-rst section, remove from the `exclude` list the file `doc/source/whatsnew/v0.15.0.rst`, `doc/source/whatsnew/v0.15.1.rst` & `doc/source/whatsnew/v0.15.2.rst`
2. Remove the import block ( `from pandas import *`) from the lines (8-11) of the file `v0.15.*.rst`. 

After that, running the next command will report the errors in the file (note that syntax error usually prevent to validate other errors, and the list of errors to fix can become much longer when the syntax error is fixed (please make sure that you are using `flake8-rst` version 0.7.0 or higher):

```
$ flake8-rst doc/source/whatsnew/v0.15.*
doc/source/whatsnew/v0.15.0.rst:75:10: F821 undefined name 'DataFrame'
doc/source/whatsnew/v0.15.0.rst:75:25: E231 missing whitespace after ':'
doc/source/whatsnew/v0.15.0.rst:75:28: E231 missing whitespace after ','
doc/source/whatsnew/v0.15.0.rst:75:30: E231 missing whitespace after ','
doc/source/whatsnew/v0.15.0.rst:75:32: E231 missing whitespace after ','
doc/source/whatsnew/v0.15.0.rst:75:34: E231 missing whitespace after ','
doc/source/whatsnew/v0.15.0.rst:75:36: E231 missing whitespace after ','
doc/source/whatsnew/v0.15.0.rst:75:52: E231 missing whitespace after ':'
doc/source/whatsnew/v0.15.0.rst:75:84: E501 line too long (80 > 79 characters)
doc/source/whatsnew/v0.15.0.rst:84:84: E501 line too long (96 > 79 characters)
doc/source/whatsnew/v0.15.0.rst:121:21: F821 undefined name 'Timedelta'
...
doc/source/whatsnew/v0.15.1.rst:26:10: F821 undefined name 'Series'
doc/source/whatsnew/v0.15.1.rst:26:17: F821 undefined name 'date_range'
doc/source/whatsnew/v0.15.1.rst:26:38: E231 missing whitespace after ','
doc/source/whatsnew/v0.15.1.rst:26:48: E231 missing whitespace after ','
doc/source/whatsnew/v0.15.1.rst:65:14: F821 undefined name 'df'
doc/source/whatsnew/v0.15.1.rst:65:25: F821 undefined name 'ts'
doc/source/whatsnew/v0.15.1.rst:154:37: E231 missing whitespace after ','
...
doc/source/whatsnew/v0.15.1.rst:231:85: E501 line too long (86 > 79 characters)
doc/source/whatsnew/v0.15.2.rst:77:30: E231 missing whitespace after ':'
doc/source/whatsnew/v0.15.2.rst:152:32: F821 undefined name 'engine'
...
```

Once all the errors are addressed, please open a pull request with the fixes in the file, and removing the file from setup.cfg. If you need to do something that feels wrong to fix an error, please ask in a comment to this issue. Please avoid other unrelated changes, which can be addressed in a separate pull request.
cc: @datapythonista

### PR DESCRIPTION
**Pull request by joybh98**
- [x] closes #24239 
- [ ] tests added / passed
- [ ] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry
Fixed PEP8 errors in **doc/source/whatsnew/v0.15.***, still can't get rid of **line too long ( x >79)** where x is the number of characters in line.

### CODE REVIEW

File: doc/source/whatsnew/v0.15.0.rst [hunk 1]
```diff
@@ -7,10 +7,11 @@ v0.15.0 (October 18, 2014)
 
 .. ipython:: python
    :suppress:
-
+   
    from pandas import * # noqa F401, F403
```
Comment (datapythonista) [inline doc/source/whatsnew/v0.15.0.rst]: we want to delete this whole block, that's one of the main points of the issue (see the description)

File: doc/source/whatsnew/v0.15.0.rst [hunk 2]
```diff
@@ -7,10 +7,11 @@ v0.15.0 (October 18, 2014)
...
 
 
+
```
Comment (datapythonista) [inline doc/source/whatsnew/v0.15.0.rst]: this blank line is unnecessary, and it should break the validation

**Review by datapythonista — COMMENTED** [review]
Looks good, thanks for the PR. Some comments in the code. Besides those:
- Remove the `test.parquet` file you committed
- Edit `setup.cfg` and remove the files you fixed from the list of exclude files of flake8-rst, so they start being validated
- Edit the PR description and replace the `#xxxxx` by the issue you're closing

**Comment by joybhallaa** [comment]
Okay,will fix all the problems.

File: doc/source/whatsnew/v0.15.0.rst (hunk 1 above)
Comment (joybhallaa) [reply doc/source/whatsnew/v0.15.0.rst]: Just to be sure, you want me to remove the whole ipython block, am i right?

Comment (datapythonista) [reply doc/source/whatsnew/v0.15.0.rst]: yes, you can see a similar case here: https://github.com/pandas-dev/pandas/pull/24236/files

**Comment by jreback** [comment]
pls merge master as well

**Comment by datapythonista** [comment]
@joybhallaa can you update in the description the issue you're closing

**Comment by datapythonista** [comment]
And when you fixed the conflict, looks like the `0.14.*` files have been added back to the list of exclusions (check the diff in the PR). Can you remove them again.

Don't worry much, all these things usually happen to every first contributor. But we need everything in place to be able to merge this. :)

File: doc/source/whatsnew/v0.15.0.rst [hunk 3]
```diff
@@ -77,7 +71,7 @@ For full docs, see the :ref:`categorical introduction <categorical>` and the
 .. ipython:: python
     :okwarning:
 
-    df = DataFrame({"id":[1,2,3,4,5,6], "raw_grade":['a', 'b', 'b', 'a', 'a', 'e']})
+    df = DataFrame({"id": [1, 2, 3, 4, 5, 6], "raw_grade": ['a', 'b', 'b', 'a', 'a', 'e']})
```
Comment (datapythonista) [inline doc/source/whatsnew/v0.15.0.rst]: Is this the line too long you mention in the description? You can fix it by simply:
```
df = DataFrame({"id": [1, 2, 3, 4, 5, 6],
                "raw_grade": ['a', 'b', 'b', 'a', 'a', 'e']})
```
Or something similar, if you need to do it somewhere else too.

Comment (joybhallaa) [reply doc/source/whatsnew/v0.15.0.rst]: I tried this,but it led to more pep8 errors.

Comment (datapythonista) [reply doc/source/whatsnew/v0.15.0.rst]: If the block is a code-block, you may need to add `...` to the beginning of the continuation file. If you can't get it fixed yourself, leave it the way you think it's correct in the PR, so we can review it, and also see the errors in the CI.

**Comment by joybhallaa** [comment]
@datapythonista , no need to comment, i know there are some errors left, but my machine is giving me some problems and I'll try to push the changes ASAP and will keep pushing them till all tests are passed.

**Comment by joybhallaa** [comment]
@datapythonista , can't get rid of the remaining errors.

File: doc/source/whatsnew/v0.15.0.rst [hunk 4]
```diff
@@ -836,7 +833,7 @@ Other notable API changes:
 
   .. code-block:: python
```
Comment (datapythonista) [inline doc/source/whatsnew/v0.15.0.rst]: This indentation is wrong, should be at the top level. And this should be a `.. code-block:: ipython` (ipython instead of python.

See similar examples for the exact format.

File: doc/source/whatsnew/v0.15.1.rst [hunk 5]
```diff
@@ -28,7 +22,7 @@ API changes
 
   .. ipython:: python
 
-     s = Series(date_range('20130101',periods=5,freq='D'))
+     s = pd.Series(pd.date_range('20130101', periods=5, freq='D'))
      s.iloc[2] = np.nan
      s
```
Comment (datapythonista) [inline doc/source/whatsnew/v0.15.1.rst]: can you try to redefine `df` and `ts` before line 64 (can't comment there, as it's not edited in the PR)

I think they should be accessible from the previous block. You can also use `# noqa F821` in the failing line, to stop the error from being raised.

**Review by datapythonista — COMMENTED** [review]
Besides the comments, can you restore the `style.ipynb` you deleted please

**Comment by datapythonista** [comment]
You've got whitespaces in `doc/source/whatsnew/v0.15.0.rst` line 729, if you can remove them (and it can be a good idea to activate validation of pep8 errors in general in your editor, to avoid those).

Also, if you can merge master.

File: doc/source/whatsnew/v0.15.0.rst [hunk 6]
```diff
@@ -724,61 +722,61 @@ Other notable API changes:
...
 - Assigning values to ``None`` now considers the dtype when choosing an 'empty' value (:issue:`7941`).
+         print("KeyError: " + str(e))
+
 
   Previously, assigning to ``None`` in numeric containers changed the
   dtype to object (or errored, depending on the call). It now uses
   ``NaN``:
 
-  .. ipython:: python
+  ..  ipython:: python
```
Comment (datapythonista) [inline doc/source/whatsnew/v0.15.0.rst]: there should be only space between the `..` and `ipython`. Here, and in all the cases that follow.

File: doc/source/whatsnew/v0.15.0.rst [hunk 7]
```diff
@@ -86,7 +81,8 @@ For full docs, see the :ref:`categorical introduction <categorical>` and the
     df["grade"].cat.categories = ["very good", "good", "very bad"]
 
     # Reorder the categories and simultaneously add the missing categories
-    df["grade"] = df["grade"].cat.set_categories(["very bad", "bad", "medium", "good", "very good"])
+    df["grade"] = df["grade"].cat.set_categories(
+      ["very bad", "bad", "medium", "good", "very good"])
```
Comment (datapythonista) [inline doc/source/whatsnew/v0.15.0.rst]: better something like:
```
df["grade"] = df["grade"].cat.set_categories(["very bad",
                                              "bad",
...
```

**Review by datapythonista — CHANGES_REQUESTED** [review]
couple of comments, besides the problem with the whitespaces already mentioned

@joybhallaa do you have time to finish this?

File: doc/source/whatsnew/v0.15.0.rst (hunk 6 above)
Comment (joybhallaa) [reply doc/source/whatsnew/v0.15.0.rst]: Yes, i will work on it ASAP.

**Comment by jreback** [comment]
can you merge master and update

**Comment by jreback** [comment]
you have an error:https://dev.azure.com/pandas-dev/pandas/_build/results?buildId=6317

**Comment by joybhallaa** [comment]
@datapythonista in v0.15.0,all **pd,df** are giving invalid syntax error,how can this be removed ?

**Comment by jorisvandenbossche** [comment]
Note there are still a bunch of failures in the doc build related to the clean-up (see eg https://travis-ci.org/pandas-dev/pandas/jobs/474277960)

File: doc/source/whatsnew/v0.15.0.rst [hunk 8]
```diff
@@ -318,11 +316,12 @@ Timezone handling improvements
   .. ipython:: python
      :okwarning:
 
-     ts = Timestamp('2014-08-01 09:00', tz='US/Eastern')
+     ts = pd.Timestamp('2014-08-01 09:00', tz='US/Eastern')
      ts
      ts.tz_localize(None)
 
-     didx = DatetimeIndex(start='2014-08-01 09:00', freq='H', periods=10, tz='US/Eastern')
+     didx = pd.DatetimeIndex(
```
Comment (datapythonista) [inline doc/source/whatsnew/v0.15.0.rst]: better:
```
didx = pd.DatetimeIndex(start='2014-08-01 09:00',
...
```

File: doc/source/whatsnew/v0.15.0.rst [hunk 9]
```diff
@@ -86,7 +81,10 @@ For full docs, see the :ref:`categorical introduction <categorical>` and the
     df["grade"].cat.categories = ["very good", "good", "very bad"]
 
     # Reorder the categories and simultaneously add the missing categories
-    df["grade"] = df["grade"].cat.set_categories(["very bad", "bad", "medium", "good", "very good"])
+    df["grade"] = df["grade"].cat.set_categories(["very bad",
+                                                  "bad", "medium",
+                                                 "good", "very good"]
```
Comment (datapythonista) [inline doc/source/whatsnew/v0.15.0.rst]: this indentation needs to be at the same level as the previos. The `)` in the next line should be at the end of this.

File: doc/source/whatsnew/v0.15.0.rst [hunk 10]
```diff
@@ -235,9 +233,9 @@ A new display option ``display.memory_usage`` (see :ref:`options`) sets the defa
     dtypes = ['int64', 'float64', 'datetime64[ns]', 'timedelta64[ns]',
               'complex128', 'object', 'bool']
     n = 5000
-    data = dict([ (t, np.random.randint(100, size=n).astype(t))
-                    for t in dtypes])
-    df = DataFrame(data)
+    data = [(t, np.random.randint(100, size=n).astype(t))
```
Comment (datapythonista) [inline doc/source/whatsnew/v0.15.0.rst]: data must be a dict, restore the previous

File: doc/source/whatsnew/v0.15.0.rst [hunk 11]
```diff
@@ -503,21 +502,24 @@ Rolling/Expanding Moments improvements
...
+     In[7]: pd.ewma(pd.Series([None, 1., 8.]), com=2.)
      Out[7]:
      0    NaN
      1    1.0
      2    5.2
      dtype: float64
 
-     In [8]: pd.ewma(Series([1., None, 8.]), com=2., ignore_na=True)  # pre-0.15.0 behavior
+     In [8]: pd.ewma(pd.Series
+             ([1., None, 8.]), com=2., ignore_na=True)  # pre-0.15.0 behavior
```
Comment (datapythonista) [inline doc/source/whatsnew/v0.15.0.rst]: restore the previous

File: doc/source/whatsnew/v0.15.0.rst [hunk 12]
```diff
@@ -503,21 +502,24 @@ Rolling/Expanding Moments improvements
...
      Out[8]:
      0    1.0
      1    1.0
      2    5.2
      dtype: float64
 
-     In [9]: pd.ewma(Series([1., None, 8.]), com=2., ignore_na=False)  # new default
+     In[9]: pd.ewma(
```
Comment (datapythonista) [inline doc/source/whatsnew/v0.15.0.rst]: restore the previous

File: doc/source/whatsnew/v0.15.0.rst [hunk 13]
```diff
@@ -612,8 +614,8 @@ Improvements in the sql io module
 
   .. code-block:: python
 
-         df.to_sql('table', engine, schema='other_schema')
-         pd.read_sql_table('table', engine, schema='other_schema')
+         df.to_sql('table', pd.engine, schema='other_schema')
+         pd.read_sql_table('table', pd.engine, schema='other_schema')
```
Comment (datapythonista) [inline doc/source/whatsnew/v0.15.0.rst]: those are incorrect, remove them

File: doc/source/whatsnew/v0.15.0.rst [hunk 14]
```diff
@@ -754,32 +756,32 @@ Other notable API changes:
...
-     s2 = s
-     s += 1.5
+      s = pd.Series([1, 2, 3])
+      s2 = s
+      s += 1.5
 
   Behavior prior to v0.15.0
 
-  .. code-block:: ipython
+  ..  code-block:: ipython
```
Comment (datapythonista) [inline doc/source/whatsnew/v0.15.0.rst]: I think I already said it in the previous review. All those spaces added here and next are wrong, restore the previous code (everything until the end of the file, excepting when you add `pd.`).

File: doc/source/whatsnew/v0.15.1.rst [hunk 15]
```diff
@@ -23,12 +17,16 @@ users upgrade to this version.
 
 API changes
 ~~~~~~~~~~~
+  .. ipython:: python
+      :supress:
+      df = pd.DataFrame()
+      ts = pd.Timestamp()
```
Comment (datapythonista) [inline doc/source/whatsnew/v0.15.1.rst]: remove this for now.

File: doc/source/whatsnew/v0.15.1.rst [hunk 16]
```diff
@@ -116,8 +114,8 @@ API changes
 
   previous behavior:
 
-  .. code-block:: ipython
 
+  .. code-block:: ipython
```
Comment (datapythonista) [inline doc/source/whatsnew/v0.15.1.rst]: restore the original

File: doc/source/whatsnew/v0.15.1.rst [hunk 17]
```diff
@@ -233,7 +231,8 @@ Enhancements
 
   .. ipython:: python
 
-     dfi = DataFrame(1,index=pd.MultiIndex.from_product([['a'],range(1000)]),columns=['A'])
+     dfi = pd.DataFrame(
+         1, index=pd.MultiIndex.from_product([['a'], range(1000)]), columns=['A'])
```
Comment (datapythonista) [inline doc/source/whatsnew/v0.15.1.rst]: restore the original, fix the missing spaces only

File: doc/source/whatsnew/v0.15.2.rst [hunk 18]
```diff
@@ -154,7 +148,7 @@ Other enhancements:
   .. code-block:: python
 
      from sqlalchemy.types import String
-     data.to_sql('data_dtype', engine, dtype={'Col_1': String})
+     data.to_sql('data_dtype', pd.engine, dtype={'Col_1': String})
```
Comment (datapythonista) [inline doc/source/whatsnew/v0.15.2.rst]: restore the original

File: setup.cfg [hunk 19]
```diff
@@ -46,9 +46,14 @@ ignore = E402,  # module level import not at top of file
...
-    doc/source/whatsnew/v0.15.1.rst
-    doc/source/whatsnew/v0.15.2.rst
+
+    doc/source/whatsnew/v0.13.0.rst
+    doc/source/whatsnew/v0.13.1.rst
+    doc/source/whatsnew/v0.16.0.rst
+    doc/source/whatsnew/v0.16.1.rst
+    doc/source/whatsnew/v0.16.2.rst
+    doc/source/whatsnew/v0.17.0.rst
+    doc/source/whatsnew/v0.17.1.rst
```
Comment (datapythonista) [inline setup.cfg]: restore the original, don't leave any changes to this file

**Review by datapythonista — CHANGES_REQUESTED** [review]
Forget about the syntax errors, and any previous comment. Just fix the comments in this review, we'll merged the changes, and anything remaining will be addressed in a separate PR.

File: doc/source/whatsnew/v0.15.0.rst [hunk 20]
```diff
@@ -255,12 +252,12 @@ Additionally :meth:`~pandas.DataFrame.memory_usage` is an available method for a
 ^^^^^^^^^^^^
 
 ``Series`` has gained an accessor to succinctly return datetime like properties for the *values* of the Series, if its a datetime/period like Series. (:issue:`7207`)
-This will return a Series, indexed like the existing Series. See the :ref:`docs <basics.dt_accessors>`
 
+This will return a Series, indexed like the existing Series. See the :ref:`docs <basics.dt_accessors>`
 .. ipython:: python
```
Comment (jorisvandenbossche) [inline doc/source/whatsnew/v0.15.0.rst]: Can you revert this change? (there needs to be a blank line between the last sentence and `.. ipython::`)

File: doc/source/whatsnew/v0.15.0.rst [hunk 21]
```diff
@@ -503,21 +502,22 @@ Rolling/Expanding Moments improvements
...
+     In [8]: pd.ewma(pd.Series([1., None, 8.]), com=2., ignore_na=True)  # pre-0.15.0 behavior
      Out[8]:
      0    1.0
      1    1.0
      2    5.2
      dtype: float64
 
-     In [9]: pd.ewma(Series([1., None, 8.]), com=2., ignore_na=False)  # new default
+     In[9]: pd.ewma(pd.Series([1., None, 8.]), com=2., ignore_na=False)  # new default
+
```
Comment (jorisvandenbossche) [inline doc/source/whatsnew/v0.15.0.rst]: Can you revert this change?

File: doc/source/whatsnew/v0.15.0.rst [hunk 22]
```diff
@@ -725,27 +725,27 @@ Other notable API changes:
...
+     s = pd.Series(np.arange(3, dtype='int64'),
+                   index=pd.MultiIndex.from_product([['A'], ['foo', 'bar', 'baz']],
+                                                    names=['one', 'two'])
+                   ).sort_index()
+
      try:
-        s.loc[['D']]
+         s.loc[['D']]
      except KeyError as e:
-        print("KeyError: " + str(e))
```
Comment (jorisvandenbossche) [inline doc/source/whatsnew/v0.15.0.rst]: Can you add back this print statement?

File: doc/source/whatsnew/v0.15.0.rst [hunk 23]
```diff
@@ -792,13 +791,13 @@ Other notable API changes:
 
   This is now the correct behavior
 
-  .. ipython:: python
+  ..  ipython:: python
```
Comment (jorisvandenbossche) [inline doc/source/whatsnew/v0.15.0.rst]: revert the addition of the space here (sphinx is very sensitive to this kind of things ..)

File: doc/source/whatsnew/v0.15.0.rst [hunk 24]
```diff
@@ -814,13 +813,13 @@ Other notable API changes:
   In prior versions this would drop the timezone, now it retains the timezone,
   but gives a column of ``object`` dtype:
 
-  .. ipython:: python
+  ..  ipython:: python
```
Comment (jorisvandenbossche) [inline doc/source/whatsnew/v0.15.0.rst]: same here, and in a couple of ones below

File: doc/source/whatsnew/v0.15.0.rst [hunk 25]
```diff
@@ -835,9 +834,9 @@ Other notable API changes:
 
 - ``SettingWithCopy`` raise/warnings (according to the option ``mode.chained_assignment``) will now be issued when setting a value on a sliced mixed-dtype DataFrame using chained-assignment. (:issue:`7845`, :issue:`7950`)
 
-  .. code-block:: python
+ ..  code-block:: ipython
```
Comment (jorisvandenbossche) [inline doc/source/whatsnew/v0.15.0.rst]: remove space

File: doc/source/whatsnew/v0.15.1.rst [hunk 26]
```diff
@@ -263,8 +256,8 @@ Enhancements
 
 - Added support for 3-character ISO and non-standard country codes in :func:`io.wb.download()` (:issue:`8482`)
 
-- World Bank data requests now will warn/raise based
   on an ``errors`` argument, as well as a list of hard-coded country codes and
+- World Bank data requests now will warn/raise based
```
Comment (jorisvandenbossche) [inline doc/source/whatsnew/v0.15.1.rst]: undo this switching of the lines

**Comment by datapythonista** [comment]
Closing this, it's been like 4 or 5 times that we request the same changes in the reviews. This is taking to much of our time to review.
